### ISSUE TITLE
Reviewing pull requests

### ISSUE THREAD

**Issue by nassimberrada** [unifyai/ivy#6542]
### General steps

When reviewing pull requests (PRs), there are a couple of things to watch out for:

1. When a PR is first created, someone from our Research Engineer team will be randomly assigned to the PR, but the author of the PR cannot directly request a review from the PR assignee. Therefore, you should always first filter for PRs with:
        Reviews -> “Not reviewed by you”
        Assignee -> your account
    These are the most important PRs to give attention, as they have received no feedback 
2. You should always start on the oldest PR first, you can sort for this on GitHub
3. If there are no PRs which meet these two criteria, then you should filter for:
        Reviews -> “Awaiting review from you”
        Assignee -> your account
    If there are also no PRs meeting this criteria, then you have no PRs to review!
4. If the PR is for a bullet point in a ToDo list issue, then check if the PR author has commented on the issue with “Close #Issue_number”. If they have not added this comment, then you should comment “Hey <@their_github_id> 🙂, I will link this to the corresponding issue, but in the future you should do this yourself by adding a comment ‘Close #Issue_number’, as explained [here](https://lets-unify.ai/ivy/contributing/1_the_basics.html#todo-list-issues)”
5. Link the associated issue to the PR manually if the comment has not been added by the author. This can be done on the right-hand pane at the bottom, under “Development”. This ensures that the issue is closed upon the PR being merged, which in turn will mark the item as complete in the ToDo list issue. If the author has commented “Close #Issue_number”, then the PR should have already been correctly linked, but you should still verify that this is the case.
6. Add the correct labels to the PR from the selection of: Array API, Function Reformatting, NumPy Frontend, TensorFlow Frontend, PyTorch Frontend, JAX Frontend, Ivy Functional API. This is very important, and is needed in order for CI to run on the PR.
7. Make sure the commit does not include any changes in the IDE configuration, for example changes to .idea folder. The changes should all be relevant only to the function being worked on.
8. Make sure the formatting is consistent with the formatting in the [contributor guide](https://lets-unify.ai/ivy/contributing.html).
9. It is often necessary to resolve conflicts. If these cannot be resolved in the browser then you will need to clone the fork, resolve in your own terminal, and then push to the fork. The easiest way is to just run:
        ./merge_with_upstream.sh name_of_branch.

    This will either complete the merge and then push, or it will fail with conflicts for you to resolve manually. In the latter case, 
    after resolving manually, you’ll just need to do:

        git add -A
        git commit -m “some message”
        git push
10. When the PR is ready to be merged, you should choose the option “Squash and merge”. Please do not select “Create a merge commit” or “Rebase and merge”.
11. If the tests are failing, but this is not the fault of the contributor for this PR, then we should still accept this PR.
12. If the PR overlaps with some work we have been doing internally, this is not the fault of the PR author, and so we should still merge the PR, and then we can simply incorporate our own changes on top of theirs if needed. We should not ask them to choose another task, and we should not close the PR because of this overlap.
13. You should not need to hand-hold at all when reviewing PRs. If the author of the PR asks questions like “what should I do now?”, then you can simply reply “please read the contributor guide and Deep Dive sections to see what else is needed”, with links provided in your response. If their PR contains many errors, then feel free to provide a response with one or two sentences quickly explaining the main issues, and then direct them to the docs. You should not spend time painstakingly going through a poor PR, and request a ton of required changes for each line! It shouldn’t demand too much of your own valuable time !
14. The bullet-point style commit history in the body should all be deleted. For each person involved in reviewing or commenting on the PR, you should then add:
        Co-authored-by: your_github_name <your_github_email>

      for this person on a new line. The exception to this rule is Dan's GitHub account (djl11), who doen't want to be listed as 
      co-author on PRs regardless of his involvement (He has enough commits to his name already 😅). These lines should be 
      added into the now empty body (following the deletion of all content). Once merged, all names will then appear as co- 
      authors of the commit.
      In cases where multiple people review and/or contribute commits towards the same PR, then you should list each person 
      side-by-side in the Co-authored by message, like so:
          Co-authored-by: your_github_name <your_github_email>, their_github_name <their_github_email>, … etc.
15. In order to avoid the PR appearing in the [PR Dashboard](https://docs.google.com/spreadsheets/d/10OauJqCgzuUD9Nkk8paGRcJIzu4jmXEh-X2AHlmSFIo/edit?usp=sharing), you should make sure to always respond to the authors messages in some form, so that they are never the last person to message on the PR (which will always cause it to appear in the dashboard). If they have said something which doesn’t warrant a response such as “okay, I will work on it”, then you can simply reply with a 👍 to their message, and this will register as a response. This will keep your PRs away from the dashboard, and keep you in the clear so none of us need to chase after you !
